File: backend/Models/Designation.cs
using System;

namespace backend.Models
{
    public class Designation
    {
        public int Id { get; set; }
        public int OrganizationId { get; set; }
        public string? OrganizationName { get; set; }
        public string? DesignationName { get; set; }
        public int? ParentDesignationId { get; set; }
        public string? ParentDesignationName { get; set; }

        // Aliases for Dapper mapping from stored procedures (Dapper needs both get and set for mapping to work reliably in some cases)
        public string? designation_name { get => DesignationName; set => DesignationName = value; }
        public string? organization_name { get => OrganizationName; set => OrganizationName = value; }
        public string? parent_designation_name { get => ParentDesignationName; set => ParentDesignationName = value; }
        public string? designation { get => DesignationName; set => DesignationName = value; }
        public string? parent_designation { get => ParentDesignationName; set => ParentDesignationName = value; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }

    public class CreateDesignationRequest
    {
        public int OrganizationId { get; set; }
        public string OrganizationName { get; set; } = string.Empty;
        public string DesignationName { get; set; } = string.Empty;
        public int? ParentDesignationId { get; set; }
    }
}

File: flutter_frontend/lib/models/designation.dart

class Designation {
  final int id;
  final int organizationId;
  final String? organizationName;
  final String designationName;
  final int? parentDesignationId;
  final String? parentDesignationName;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Designation({
    required this.id,
    required this.organizationId,
    this.organizationName,
    required this.designationName,
    this.parentDesignationId,
    this.parentDesignationName,
    this.createdAt,
    this.updatedAt,
  });

  factory Designation.fromJson(Map<String, dynamic> json) {
    return Designation(
      id: json['id'] ?? 0,
      organizationId: json['organizationId'] ?? 0,
      organizationName: json['organizationName'],
      designationName: json['designationName'] ?? '',
      parentDesignationId: json['parentDesignationId'],
      parentDesignationName: json['parentDesignationName'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'organizationId': organizationId,
      'organizationName': organizationName,
      'designationName': designationName,
      'parentDesignationId': parentDesignationId,
      'parentDesignationName': parentDesignationName,
    };
  }
}
